### PR TITLE
feat(torii): Support graphql subscriptions

### LINKED ISSUE #484: [Torii] Support graphql subscriptions
Add support to subscribe to state updates. Examples of what entity and component subscriptions could potentially look like:

```
subscription EntitySubscription($id: ID!) {
    entitySubscribe(id: $id) {
        entity {
            id
            keys
            components: [ComponentUnion]
        }
    }
}

subscription ComponentSubscription($name: String!) {
    componentSubscribe(name: $name) {
        component {
            ...
        }
    }
}
```

Torii's graphql uses [async-graphql](https://github.com/async-graphql/async-graphql) with dynamic schema. Unfortunately, documentation for dynamic schema is a bit lacking and will require a good deal of [source code](https://github.com/async-graphql/async-graphql/blob/master/src/dynamic/subscription.rs#L276) exploration.

### PR DESCRIPTION
**Pull request by gianmarcoalarcon**
Implement dynamic subscription for entities and components

Closes #484

### CODE REVIEW

File: crates/torii/core/Cargo.toml [hunk 1]
```diff
@@ -10,10 +10,14 @@ version.workspace = true
 
 [dependencies]
 anyhow.workspace = true
+async-graphql = { version = "5.0.8", features = [ "chrono", "dynamic-schema" ] }
 async-trait.workspace = true
 chrono.workspace = true
 dojo-types = { path = "../../dojo-types" }
 dojo-world = { path = "../../dojo-world" }
+indexmap = "1.9.3"
+torii-graphql = { path = "../graphql" }
```
[the remark this answers is not in the document]
Comment (gianmarcoalarcon) [reply crates/torii/core/Cargo.toml:19]: We have to build the message that is gonna be published with the objects from graphql dependency. 
LI will try to find the way build the message without this dependency.

File: crates/torii/core/Cargo.toml [hunk 2]
```diff
@@ -10,10 +10,12 @@ version.workspace = true
 
 [dependencies]
 anyhow.workspace = true
+async-graphql = { version = "5.0.8", features = [ "chrono", "dynamic-schema" ] }
```
Comment (tarrencev) [inline crates/torii/core/Cargo.toml:13]: is this necessary? core shouldn't know about graphql

Comment (gianmarcoalarcon) [reply crates/torii/core/Cargo.toml]: Fixed on commit 6d5b936

**Comment by broody** [comment]
Just few suggestions otherwise looks good! Please also add a test for this

**Comment by gianmarcoalarcon** [comment]
@broody Script test implement on commit [96b4193]

**Comment by broody** [comment]
> @broody Script test implemented on commit [96b4193](https://github.com/dojoengine/dojo/commit/96b41938bae0089f64991300df2ec6d335caab26)

Forgot to submit my reviews... my bad! but shouldn't effect the tests tho just what's being published

File: crates/torii/graphql/src/schema.rs [hunk 3]
```diff
@@ -82,7 +90,33 @@ pub async fn build_schema(pool: &SqlitePool) -> Result<Schema> {
         schema_builder = schema_builder.register(object.create());
     }
 
-    schema_builder.register(query_root).data(pool.clone()).finish().map_err(|e| e.into())
+    // todo: find a way to iterate over fields to create arguments
+    let subscription_root = Subscription::new("Subscription")
```
[the remark this answers is not in the document]
Comment (gianmarcoalarcon) [reply crates/torii/graphql/src/schema.rs]: Refactored on commit 9922548

**Comment by broody** [comment]
Works well on my side! Made some small changes to naming to conform to graphql conventions. Like we discussed, there seems to be some difficulty moving the query into SubscriptionField, I can take a look and refactor this later on.

**Review by broody — APPROVED** [review]

**Pull request merged**